src/commonMain/kotlin/fr/acinq/eclair/db/PaymentsDb.kt

```diff
@@ -182,17 +201,44 @@ data class OutgoingPayment(val id: UUID, val recipientAmount: MilliSatoshi, val
         /** Swaps out send a lightning payment to a swap server, which will send an on-chain transaction to a given address. */
         data class SwapOut(val address: String, override val paymentHash: ByteVector32) : Details()
 
+        /** Corresponds to the on-chain payments made when closing a channel. */
+        data class ChannelClosing(
+            val channelId: ByteVector32,
+            val closingAddress: String,
+            val isSentToMyWallet: Boolean,
+            override val paymentHash: ByteVector32
+        ) : Details()
+
         fun matchesFilters(filters: Set<PaymentTypeFilter>): Boolean = when (this) {
             is Normal -> filters.isEmpty() || filters.contains(PaymentTypeFilter.Normal)
             is KeySend -> filters.isEmpty() || filters.contains(PaymentTypeFilter.KeySend)
             is SwapOut -> filters.isEmpty() || filters.contains(PaymentTypeFilter.SwapOut)
+            is ChannelClosing -> filters.isEmpty() || filters.contains(PaymentTypeFilter.ChannelClosing)
         }
     }
 
     sealed class Status {
         object Pending : Status()
-        data class Succeeded(val preimage: ByteVector32, val completedAt: Long = currentTimestampMillis()) : Status()
-        data class Failed(val reason: FinalFailure, val completedAt: Long = currentTimestampMillis()) : Status()
+        sealed class Completed : Status() {
+            abstract val completedAt: Long
+            data class Failed(val reason: FinalFailure, override val completedAt: Long = currentTimestampMillis()) : Completed()
+            sealed class Succeeded : Completed() {
+                data class OffChain(
+                    val preimage: ByteVector32,
+                    override val completedAt: Long = currentTimestampMillis()
+                ) : Succeeded()
+                data class OnChain(
+                    val txids: List<ByteVector32>,
+                    val claimed: Satoshi,
+                    val type: ChannelClosingType,
+                    override val completedAt: Long = currentTimestampMillis()
+                ) : Succeeded() {
+                    enum class ChannelClosingType {
+                        Mutual, Local, Remote, Other
```

Comment:
Why don't you support `Revoked` here? It looks like it would be trivial to do, wouldn't it?

Reply:
Indeed. Done in 202dfc2